src/hooks/useColumns.tsx

```diff
@@ -167,17 +204,17 @@ function useColumns<RecordType>(
         },
       ];
     }
-
     return finalColumns;
-  }, [transformColumns, withExpandColumns]);
-
-  const flattenColumns = React.useMemo(() => flatColumns(mergedColumns), [mergedColumns]);
+  }, [transformColumns, withExpandColumns, direction]);
 
+  const flattenColumns = React.useMemo(
+    () => reverseFixedForRtlDirection(flatColumns(mergedColumns, direction), direction),
+    [mergedColumns],
```

Comment:
do we need add `direction` into the condition also?

Reply:
I made some changes to that class.